### PR TITLE
feat(utils): updated email verification with more RFC 5322-compliant …

### PR DESCRIPTION
**Pull request by mailwilliams**
…check, added phone number check

Issues:

- https://linear.app/metriport/issue/ENG-1036

### Dependencies

- Upstream: None
- Downstream: https://github.com/metriport/metriport-internal/pull/3246

### Description

Zod's email check isn't a fully comprehensive check (article from Zod's maintainer [here](https://colinhacks.com/essays/reasonable-email-regex)), so replacing it with a more comprehensive custom regex catches more valid emails. Being fully RFC5322 compliant is unreasonable, but this is an improvement.

- Remove zod's `string().email().safeParse`
- Replaced with upgraded regex to match more valid emails
  - Handles `!` in email edge case
  - Handles phone numbers being sent as emails (starting with `+1`) 
- Added more comprehensive testing 

### Testing

- Local
  - [x] Ran updated test suite

Check each PR.

### Release Plan

This update (given existing unit tests) should be backwards compatible with what we currently have, and if we are notified of new errors we can roll back the change

- [x] No dependencies between API and Infra that will cause errors during deployment
- [ ] Merge this

## Summary by CodeRabbit

* **New Features**
  * Added a public phone-number detection utility and applied broader email normalization/validation across the SDK (custom email validation used in schema).

* **Bug Fixes**
  * Prevented phone-number-like inputs (e.g., +1-prefixed) from being accepted as emails and added clearer, phone-specific error messages.
  * Improved handling of whitespace, mailto: prefixes, special characters, and other edge cases.

* **Tests**
  * Expanded coverage for email formats, phone-number detection, normalization behavior, and fixed test descriptions.

### CODE REVIEW

File: packages/shared/src/domain/contact/email.ts [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
 
 export function isEmailValid(email: string): boolean {
   if (!email) return false;
   if (email.length === 0) return false;
-  const safeParseEmail = z.string().email().safeParse(email);
-  if (!safeParseEmail.success) return false;
-  return true;
+
+  // Check for phone number format (+1 prefix)
+  if (email.startsWith("+1")) {
```
Comment (keshavsaharia) [inline packages/shared/src/domain/contact/email.ts:19]: The email address `+1person@example.com` is an RFC 5322-compliant email address, but would fail from this check here.

File: packages/shared/src/domain/contact/email.ts [hunk 2]
```diff
@@ -1,19 +1,34 @@
-import { z } from "zod";
 import { BadRequestError } from "../../error/bad-request";
 
 export const exampleEmail = "test@test.com";
 const mailtoPrefix = "mailto:";
 
+// Enhanced email validation that properly handles RFC 5322 characters
+const EMAIL_REGEX =
+  /^(?=.{1,254}$)(?=.{1,64}@)[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[A-Za-z0-9](?:[A-Za-z0-9-]{0,61}[A-Za-z0-9])?\.)+[A-Za-z]{2,63}$/;
```
Comment (keshavsaharia) [inline packages/shared/src/domain/contact/email.ts:8]: Would you be able to add the source for this regex in the comments?

File: packages/shared/src/domain/contact/email.ts (hunk 1 above)
Comment (mailwilliams) [reply packages/shared/src/domain/contact/email.ts:19]: @keshavsaharia correct, I spoke with @leite08 and basically the line has to be drawn somewhere. We have agreed that we are assuming a `+1...` email is an error, and if this becomes an issue we can look into more edge cases to allow

File: packages/shared/src/domain/contact/email.ts (hunk 2 above)
Comment (mailwilliams) [reply packages/shared/src/domain/contact/email.ts:8]: @keshavsaharia  Thanks for flagging. This was generated to whitelist more RFC-5322-compliant features, but I will add the translation in plain English and find the matching information in the 5322 docs
